app/routes.py

```diff
@@ -1,2 +1,247 @@
+    return "", 404
+
+@video_bp.route("/<video_id>", methods=["PUT"], strict_slashes=False)
+def update_video(video_id):
+    video = Video.query.get(video_id)
+    if video:
+        request_body = request.get_json()
+        if "title" in request_body and "release_date" in request_body and "total_inventory" in request_body:
+            video.title = request_body["title"]
+            video.release_date = request_body["release_date"]
+            video.total_inventory = request_body["total_inventory"]
+            db.session.commit()
+            updated_video = video.to_json()
+            return jsonify(updated_video), 200
+        return jsonify({
+        "details": "Invalid data"
+        }), 400
+    return "", 404
+
+@video_bp.route("/<video_id>", methods=["DELETE"], strict_slashes=False)
+def delete_video(video_id):
+    video = Video.query.get(video_id)
+    if video:
+        db.session.delete(video)
+        db.session.commit()
+        return jsonify({
+        "id": video.video_id
+        }), 200
+    return "", 404
+
+@rental_bp.route("/check-out", methods=["POST"], strict_slashes=False)
+def add_rental():
+    request_body = request.get_json()
+    if "customer_id" not in request_body or "video_id" not in request_body:
+        return jsonify({
+        "details": "Invalid data"
+        }), 400
+
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+    try:
+        customer = Customer.query.get(customer_id)
+    except:
+        return jsonify({
+        "details": "no such customer exists"
+        }), 400
+
+    try:
+        video = Video.query.get(video_id)
+    except:
+        return jsonify({
+        "details": "no such video exists"
+        }), 400
+
+    customer.videos_checked_out_count += 1
+    if video.available_inventory < 1:
+        return jsonify({
+        "details": "not enough inventory available. hehe"
+        }), 400
+    video.available_inventory -= 1
+
+    new_rental = Rental(customer_id=request_body["customer_id"],
+                            video_id=request_body["video_id"])
+    
+    db.session.add(new_rental)
+    db.session.commit()
+    return jsonify({
+        "customer_id": new_rental.customer_id,
+        "video_id" : new_rental.video_id,
+        "due_date" : new_rental.due_date,
+        "videos_checked_out_count": customer.videos_checked_out_count,
+        "available_inventory": video.available_inventory
+        }), 200
+
+
+@rental_bp.route("/check-in", methods=["POST"], strict_slashes=False)
+def remove_rental():
+    request_body = request.get_json()
+    if "customer_id" not in request_body or "video_id" not in request_body:
+                    return jsonify({
+                    "details": "Invalid data"
+                    }), 400
+
+    customer_id_request = request_body["customer_id"]
+    results = db.session.query(Customer, Video, Rental).join(Customer, Customer.customer_id==Rental.customer_id)\
```

Comment:
Instead of doing a complicated query like this you can find the rental by its customer_id and video_id and then just use `.customer` and `.video` on the returned object.